<!DOCTYPE html>
<!--

Copyright (c) 2013 Samsung Electronics Co., Ltd.

Licensed under the Apache License, Version 2.0 (the License);
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Authors:
        Beata Koziarek <b.koziarek@samsung.com>

-->

<html>
<head>
<title>MessageService_syncFolder_successCallback_invalid_cb</title>
<script type="text/javascript" src="support/unitcommon.js"></script>
<script type="text/javascript" src="support/messaging_common.js"></script>
</head>
<body>
<div id="log"></div>
<script type="text/javascript">
//==== TEST: MessageService_syncFolder_successCallback_invalid_cb
//==== LABEL Check if MessageService.syncFolder method throws exception when successCallback is invalid
//==== PRIORITY: P2
//==== ONLOAD_DELAY 300
//==== TIMEOUT 300
//==== SPEC: Tizen Web API:Communication:Messaging:MessageService:syncFolder M
//==== SPEC_URL https://developer.tizen.org/help/topic/org.tizen.web.device.apireference/tizen/messaging.html
//==== TEST_CRITERIA MTCB
setup({timeout: 300000});
var t = async_test(document.title, {timeout: 300000}),
    serviceError, serviceSuccess, service, findError, findSuccess, invalidSuccessCallback;
t.step(function () {
    invalidSuccessCallback = {
        onsuccess: t.step_func(function () {
            assert_unreached("Invalid success callback invoked");
        })
    };

    findSuccess = t.step_func(function (folders) {
        assert_greater_than(folders.length, 0, "Received empty folders array");

        assert_throws(TYPE_MISMATCH_EXCEPTION, function () {
            service.syncFolder(folders[0], invalidSuccessCallback);
        },  "TypeMismatchError should be thrown - given incorrect successCallback.");
        t.done();
    });

    findError = t.step_func(function (error) {
        assert_unreached("findFolders() error callback: name:" + error.name + ", msg:" + error.message);
    });

    serviceSuccess = t.step_func(function (emailService) {
        service = emailService;
        findFolders(t, service, findSuccess, findError);
    });

    serviceError = t.step_func(function (error) {
        assert_unreached("getEmailService() error callback: name:" + error.name + ", msg:" + error.message);
    });

    getEmailService(t, serviceSuccess, serviceError);
});

</script>
</body>
</html>
